src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,243 @@
+package duke;
+
+
+import java.time.DateTimeException;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.format.DateTimeParseException;
+import java.util.regex.Pattern;
+
+import duke.command.AddCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.ExitCommand;
+import duke.command.FindByDateCommand;
+import duke.command.FindByDescriptionCommand;
+import duke.command.ListCommand;
+import duke.command.MarkDoneCommand;
+import duke.task.Task;
+import duke.task.TaskList;
+import duke.task.TaskType;
+
+/**
+ * The Parser class is responsible for parsing all user commands and input dates and times for Duke.
+ *
+ * @author Aiken Wong
+ */
+public class Parser {
+
+    private static String errorMessage = "Wrong format Sir/Mdm. Dates and times must be given as only a date: DATE\n"
+        + "or as date and time: DATE TIME\n"
+        + "Accepted formats for DATE: YYYY-MM-DD, DD/MM/YYYY\n"

```

Comment:
Nothing wrong with this, but there's no need to end your substrings with "\n". You could have combined the substrings together to make it more compact!